### PR TITLE
[Search] Sanitize queries to Elasticsearch

### PR DESCRIPTION
**Pull request by jshoe**
- Filter everything that isn't `a-zA-Z0-9 _|`

### CODE REVIEW

**Comment by cdebourcy** [comment]
Is there a more native way of doing this? Has anyone ever encountered this problem before when using elasticsearch-rails?

**Comment by jshoe** [comment]
Looks like it's not present in that library and it's recommended to sanitize ourselves https://discuss.elastic.co/t/ruby-elasticsearch-ruby-special-characters-not-escaped-by-the-library/15187/3

**Comment by cdebourcy** [comment]
great

File: app/helpers/elasticsearch_helper.rb [hunk 1]
```diff
@@ -32,4 +34,10 @@ def taxon_search(prefix, tax_levels = TaxonCount::NAME_2_LEVEL.keys)
     end
     matching_taxa.values
   end
+
+  private
+
+  def sanitize(prefix)
+    prefix.gsub(/[^a-zA-Z0-9 _|]/, "") if prefix
```
Comment (tfrcarvalho) [inline app/helpers/elasticsearch_helper.rb]: You should add at least `.` to the mix.
Even then, I think the best solution is to escape the special characters: http://lucene.apache.org/core/3_0_3/queryparsersyntax.html#Escaping%20Special%20Characters

Comment (jshoe) [reply app/helpers/elasticsearch_helper.rb]: 👍 Ok modified this to add backslash in front of special characters. Tested that it works for ones like idseq-bench-4|

Side note, posts like this one point out that query_string is the most raw field http://www.recursion.org/query-parser/#user_input_may_contain_special_characters

**Pull request merged**